conan/tools/gnu/autotools.py

```diff
@@ -0,0 +1,239 @@
+                    if self._valid_configure_flag(varname, args, available_flags):
+                        args.append("--%s=${prefix}/%s" % (varname, DEFAULT_INCLUDE))
+                if self._valid_configure_flag("datarootdir", args, available_flags):
+                    args.append("--datarootdir=${prefix}/%s" % DEFAULT_SHARE)
+
+        with environment_append(pkg_env):
+            with environment_append(vars or self.vars):
+                command = '%s/configure %s %s' % (configure_dir, args_to_string(args),
+                                                  " ".join(triplet_args))
+                self._conanfile.output.info("Calling:\n > %s" % command)
+                self._conanfile.run(command, win_bash=self._win_bash, subsystem=self.subsystem)"""
+
+        cmd = "bash -c 'source autotoolsdeps.sh && source autotools.sh && %s/configure'" % configure_dir
+        self._conanfile.output.info("Calling:\n > %s" % cmd)
+        self._conanfile.run(cmd, win_bash=self._win_bash, subsystem=self.subsystem)
+
+    def _configure_help_output(self, configure_path):
+        from six import StringIO  # Python 2 and 3 compatible
+        mybuf = StringIO()
+        try:
+            self._conanfile.run("%s/configure --help" % configure_path, win_bash=self._win_bash,
+                                output=mybuf)
+        except ConanException as e:
+            self._conanfile.output.warn("Error running `configure --help`: %s" % e)
+            return ""
+        return mybuf.getvalue()
+
+    def _adjust_path(self, path):
+        if self._win_bash:
+            path = unix_path(path, path_flavor=self.subsystem)
+        return '"%s"' % path if " " in path else path
+
+    @staticmethod
+    def _valid_configure_flag(varname, args, available_flags):
+        return not AutoToolsBuildEnvironment._is_flag_in_args(varname, args) and \
+               varname in available_flags
+
+    @staticmethod
+    def _is_flag_in_args(varname, args):
+        flag = "--%s=" % varname
+        return any([flag in arg for arg in args])
+
+    def make(self, target=None):
+        """if not self._build_type:
+            raise ConanException("build_type setting should be defined.")
+        with environment_append(vars or self.vars):
+            str_args = args_to_string(args)
+            cpu_count_option = (("-j%s" % cpu_count(output=self._conanfile.output))
+                                if ("-j" not in str_args and "nmake" not in make_program.lower())
+                                else None)
+            self._conanfile.run("%s" % join_arguments([make_program, target, str_args,
+                                                       cpu_count_option]),
+                                win_bash=self._win_bash, subsystem=self.subsystem)"""
+
+        make_program = self._conanfile.conf["tools.gnu"].make_program
+        if make_program is None:
+            make_program = "mingw32-make" if platform.system() == "Windows" else "make"
```

Comment:
I also don't think it's always desired to enforce `mingw32-make` on Windows. it should go from the configuration.

Reply:
It is probably  a sane default in windows, the same as ``make`` is in other platforms. But we will see as we keep adding support for other subsystems, I am fine with removing it.